src/app/cli/src/coda.ml

```diff
@@ -551,12 +551,20 @@ let daemon logger =
            >>| Or_error.ok
          in
          Stream.iter
-           (Async.Scheduler.long_cycles
+           (Async_kernel.Async_kernel_scheduler.(
+              long_cycles_with_context @@ t ())
               ~at_least:(sec 0.5 |> Time_ns.Span.of_span_float_round_nearest))
-           ~f:(fun span ->
+           ~f:(fun (span, ctx) ->
+             let tm = Time_ns.Span.to_string span in
+             (* see if backtraces reveal anything about long async cycles *)
+             let metadata =
+               ("tm", `String tm)
+               :: List.mapi ctx.backtrace_history ~f:(fun ndx bt ->
```

Comment:
Minor nit, but You could just make this a json list instead of a list of metadata entries. Would make it easier to process the metadata.